test/box/gh-5958-rolled-back-replace-background-built-index.test.lua

```diff
@@ -0,0 +1,268 @@
+env = require('test_run')
+test_run = env.new()
+fiber = require('fiber')
+
+-- Create table with enough tuples to make index build/alter in background.
+s = box.schema.space.create('test', {engine = 'memtx'})
+i = s:create_index('pk',  {type='tree', parts={{1, 'uint'}}})
+for i = 1, 6000 do s:replace{i, i, "valid"} end
+
+started = false
+
+-- Array with origin tuples
+origin = {[0] = s:get{0}, [1] = s:get{1}, [2] = s:get{2}}
+
+
+
+------ Suite to test rolled back changes in build-in-background secondary index (unique and non-unique) ------
```

Comment:
Nits:
1. Too many empty lines; one line to separate test cases is enough;
2. To many '-' delimiters (at least at the end):)